test/e2e/validation_test.go

```diff
@@ -503,3 +503,65 @@ func (tc *testContext) validateUpgradeableCondition(expected meta.ConditionStatu
 	}
 	return nil
 }
+
+// runWindowsJob creates a Windows Server Job with the given `name` and execution `command`. In addition, check for
+// errors based on `expectError` and is able to delete the job if specified by `deleteJob`
+func runWindowsJob(name, command string, expectError, deleteJob bool,
+	t *testing.T, testCtx *testContext) *batch.Job {
+	winJob, err := testCtx.createWindowsServerJob(name,
+		[]string{
+			testCtx.getPowerShellExe(), "-Command", command,
+		},
+		nil)
+	require.NoError(t, err, "could not create Windows tester job")
+	if deleteJob {
+		defer testCtx.deleteJob(winJob.Name)
+	}
+	err = testCtx.waitUntilJobSucceeds(winJob.Name)
+	if expectError {
+		assert.Error(t, err, "Windows tester job expected to fail")
+	} else {
+		assert.NoError(t, err, "Windows tester job failed")
+	}
```

Comment:
Passing in the `testing.T` and making this check goes against our usual pattern of returning error from helpers and checking within the actual test.

Reply:
+1